### PR TITLE
iprules.sh into netcontrol

### LINKED ISSUE #31: Integrate ip rules into netcontrol
### Description

Since we're integrating [portail.sh](https://github.com/InsaLan/scripts-reseau/blob/nftables/portail.sh) into netcontrol, we might as well integrate [ip_rules.sh](https://github.com/InsaLan/scripts-reseau/blob/nftables/ip_rules_insalan.sh) directly into netcontrol, in a dynamic way that depends on the amount of marks and vpns.

It could fetch the number of VPNS from `/etc/hosts`, for example (see [vpn_register.py](https://github.com/InsaLan/VPNator/blob/master/vpn_register.py)).

### PR DESCRIPTION
**Pull request by awing-ding**
## Description

`iprules.sh` functions are converted to netcontrol functions

## Checklist

- [ ] I have tested the changes locally and they work as expected.
- [ ] I have tested the responsiveness of the changes and they work as expected.
- [ ] I have assigned the pull request to the appropriate reviewer(s).
- [x] I have added labels to the pull request, if necessary.

## Related Issues

Closes #31

### CODE REVIEW

**Comment by pixup1** [comment]
It would be better if iprules methods were in their own separate class (and file), as they don't really have anything to do with nftables

File: netcontrol/nft.py [hunk 1]
```diff
@@ -33,5 +34,59 @@ def _execute_nft_cmd(self, cmd: str) -> dict:
...
 
+    def _add_vpn(self, mark: int, vpn: str) -> None:
+        """Adds a VPN to the nftables rule
+
+        Args:
+            mark (int): mark value
+            vpn (str): VPN routing table name
+        """
+        try:
+            self._execute_nft_cmd(f"add rule insalan netcontrol-forward fwmark {mark} lookup {vpn}")
```
Comment (pixup1) [inline netcontrol/nft.py:45]: iprules aren't managed byt `nft` but by `ip rule` (cf [`iprules.sh`](https://github.com/InsaLan/scripts-reseau/blob/main/ip_rules_mini.sh))

Comment (awing-ding) [reply netcontrol/nft.py:45]: ohh okay, i clearly misunderstood the assignment :joy:

**Comment by awing-ding** [comment]
maybe the file should be elsewhere ?

**Comment by pixup1** [comment]
> maybe the file should be elsewhere ?

Do you mean outside of netcontrol ? I think it's fine there

**Comment by InsaLan-Serveurs** [comment]
You also need to add the rules on api start and remove them on api stop. Use the lifetime() function from main.py for that (It's not up to date on your branch, do a `git merge netcontrol`)

**Comment by awing-ding** [comment]
Ne marche pas (les commandes ip rules sont executés dans le conteneur)

**Pull request closed without merging**